tests/test_journal.py

```diff
@@ -342,3 +342,31 @@ def collect_ids():
     j_ids, w_ids = collect_ids()
     assert j_ids == {1, 3}
     assert w_ids == {2}
+
+
+def test_restore_dependencies(hq_env: HqEnv, tmp_path):
+    journal_path = os.path.join(tmp_path, "my.journal")
+    hq_env.start_server(args=["--journal", journal_path])
+    hq_env.start_worker()
+    tmp_path.joinpath("job.toml").write_text(
+        """
+[[task]]
+id = 1
+command = ["sleep", "0"]
+
+[[task]]
+id = 2
+command = ["sleep", "2"]
```

Comment:
Let's use some custom task requirements that will only be provided by the second worker in the test, to avoid race conditions (`wait_for_task_state` polls in some intervals, and in theory the task could be finished before we stop the server). Also we could check that the task was actually executed (e.g. by writing something to disk). The job could be marked as finished after restoration without the tasks actually running.

Reply:
Fixed